sw/device/lib/testing/flash_ctrl_testutils.h

```diff
@@ -217,33 +217,20 @@ bool flash_ctrl_testutils_bank_erase(dif_flash_ctrl_state_t *flash_state,
                                      uint32_t bank, bool data_only);
 
 /**
- * Determines a count value as the index of the first bit set to 1 in a word.
+ * Returns the value of a non-volatile counter in flash.
  *
- * This uses an incrementing counter implemented as a uint32_t initialized to
- * all 1 (meaning a count of zero), and on each increment operation the lowest
- * bit set is cleared. This is consistent with the capability of flash.
- *
- * @param strike_counter The address of the counter.
- * @return The bit position of the first bit set to 1.
+ * @param counter Counter ID, [0, 2].
+ * @return Value of the non-volatile counter
  */
-uint32_t flash_ctrl_testutils_get_count(uint32_t *strike_counter);
+uint32_t flash_ctrl_testutils_counter_get(size_t counter);
 
 /**
- * Increments a strike counter by clearing a bit.
- *
- * Uses a strike counter as described for flash_ctrl_testutils_get_count.
- * It is give a bit to be cleared, typically the value returned by the most
- * recent call to flash_ctrl_testutils_get_count.
- *
- * Notice if the right-most bit set to 1 is not cleared the counter will not
- * advance.
+ * Increments a non-volatile counter in flash.
  *
  * @param flash_state A flash_ctrl state handle.
- * @param strike_counter The address of the counter.
- * @param index The bit to clear.
+ * @param counter Counter ID, [0, 2].
  */
-void flash_ctrl_testutils_increment_counter(dif_flash_ctrl_state_t *flash_state,
-                                            uint32_t *strike_counter,
-                                            uint32_t index);
+void flash_ctrl_testutils_counter_increment(dif_flash_ctrl_state_t *flash_state,
```

Comment:
Could you provide the equivalent of the previous function as well, where we are given the current count and we simply zero the corresponding word. That avoids the loop, and should make the use of counters faster.There are some counters in ISRs for example.

Perhaps that would be "flash_ctrl_testutils_counter_strike_out"?

Reply:
I think, I'll name it something like `...counter_set_at_least()` and I should probably reverse the order of the current loop since IIUC there can be gaps now.

Reply:
Done, added `flash_ctrl_testutils_set_at_least()`. It looks like the only that increments counters in ISRs is `sw/device/tests/sim_dv/all_escalation_resets_test.c`, so I updated it to use this new function.